site/src/components/WorkspaceHealthBadge/WorkspaceHealthBadge.tsx

```diff
@@ -0,0 +1,30 @@
+import { Workspace } from "api/typesGenerated"
+import { Pill } from "components/Pill/Pill"
+import { FC, PropsWithChildren } from "react"
+import { ErrorIcon } from "components/Icons/ErrorIcon"
+import FavoriteIcon from "@mui/icons-material/Favorite"
+import { Maybe } from "components/Conditionals/Maybe"
+
+export type WorkspaceHealthBadgeProps = {
+  workspace: Workspace
+  className?: string
+}
+
+export const WorkspaceHealthBadge: FC<
+  PropsWithChildren<WorkspaceHealthBadgeProps>
+> = ({ workspace, className }) => {
+  return (
+    <Maybe
+      condition={["starting", "running", "stopping"].includes(
```

Comment:
Review: Should we modify the API endpoint instead and do this on the API so that `workspace.health = null` when it's not possible to know (e.g. stopped workspace)?